web/js/modules/layers/util.js

```diff
@@ -24,14 +24,27 @@ import util from '../../util/util';
   * @returns {Boolean} - True if layer is available at date, otherwise false
   */
 export function availableAtDate(def, date) {
```

Comment:
Good catch here! I had to take a closer look at this function on my branch when trying to re-implement the "hide unavailable" filtering and seeing your changes here was helpful.  There were just a few additional cases this didn't catch and, in modifying this to handle those cases, I think I found a way to do this that doesn't rely on `datesInDateRanges` at all:

```
export function availableAtDate(def, date) {
  const { dateRanges, inactive } = def;
  const startDate = def.startDate && new Date(def.startDate);
  const endDate = def.endDate && new Date(def.endDate);

  // Some vector layers
  if (!startDate && !dateRanges) {
    return true;
  }
  // set inactive in config
  if (endDate && inactive) {
    return date < endDate && date > startDate;
  }
  // no endDate may indicate ongoing
  if (startDate && !endDate) {
    if (!dateRanges) {
      return date > startDate;
    }
    const endRange = dateRanges.length - 1;
    const rangeEndDate = new Date(dateRanges[endRange].endDate);
    return date > startDate && rangeEndDate && date < rangeEndDate;
  }
}
```

If we just check to make sure that `date` is after the `startDate` and before the latest `endDate` of the last `dateRange`, I think we cover most cases.  I suppose it's still possible that a layer might not have coverage on a given date if there are significant gaps between `dateRange`s but that seems uncommon and I'm not seeing any obvious instances of this when testing on my branch but, I'd be curious to hear your thoughts on this.

Reply:
@jasontk19 I think your above revision is a good change to simplify/speed up the function by getting rid of the `datesInDateRanges` being used there. I'd have to play with it some to try and find any outlier "gotcha" layers, but at first glance looks like it will work great.